cmake/main.cmake

```diff
@@ -12,6 +12,8 @@ set(REDPANDA_DEPS_INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/deps_install
   CACHE STRING "Managed dependencies install directory")
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_UNITY_BUILD ON)
+set(CMAKE_UNITY_BUILD_BATCH_SIZE 10)
```

Comment:
This should be override-able with CMAKE_UNITY_BUILD when calling CMake. Perhaps `OFF` is a better default...

Reply:
That's a great idea. Is it also possible to guard it on the CMAKE version without bumping the minimum required?

Reply:
Do those flags need to be `CACHE`d?